pkg/test/cel/main_test.go

```diff
@@ -59,3 +60,26 @@ func TestMain(m *testing.M) {
 func ptrTo[T any](a T) *T {
 	return &a
 }
+
+func celErrorStringMatches(got, want string) bool {
+	gotL := strings.ToLower(got)
+	wantL := strings.ToLower(want)
+
+	// Starting in k8s v1.28, CEL error messages stopped adding spec and status prefixes to path names
+	wantLAdjusted := strings.ReplaceAll(wantL, "spec.", "")
+	wantLAdjusted = strings.ReplaceAll(wantLAdjusted, "status.", "")
+
+	// Enum validation messages changed in k8s v1.28:
+	// Before: must be one of ['Exact', 'PathPrefix', 'RegularExpression']
+	// After: supported values: "Exact", "PathPrefix", "RegularExpression"
+	if strings.Contains(wantLAdjusted, "must be one of") {
```

Comment:
@jpbetz do you know of any other changes on the horizon for CEL error messages? Any better ideas for how we can make tests like this both support multiple versions and ensure that we're returning useful error messages to users? As long as this set of workarounds remains small, I think this is ~OK, but would hate to turn this into a massive function.

Reply:
I'm not aware of any on the horizon. We're trying to apply a light touch to these things as we do understand Hyrum's law is in full effect.  But we'll keep a close eye out.

cc @TristonianJones @cici37 for visibility

Reply:
Those errors don't look like they're from CEL itself, but perhaps would be from some layer up the stack within K8s maybe?

Reply:
Thanks for the quick responses! Definitely get that this is Hyrum's law in practice, just wondering if there's anywhere we can easily track potential upstream changes.

Reply:
I looked more closely.

I don't believe the enum error is from CEL specifically. Although it is from openapi so still under api-machinery. This might be more durably matched by checking the error type, which consistently be [`FieldValueNotSupported`](https://github.com/kubernetes/kubernetes/blob/7b3589151285716cd7b0a002bab9f73c32c286df/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L121).  We do sometimes change error strings, but the error types are fixed.

The removal of status and spec prefixes on path names _surprising_, and does not look correct. @cici37 would you look more closely at this issue?